### PR TITLE
feat: added stats component to profile page

### PR DESCRIPTION
**Pull request by jnahan**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Related to https://github.com/freeCodeCamp/freeCodeCamp/issues/48234

Added stats component to profile page, removed streaks from heat map

Screen Shot 2024-04-04 at 12 33 44 PM
Screen Shot 2024-04-04 at 12 34 03 PM

### CODE REVIEW

**Comment by jenna5376** [comment]
@ahmaxed would love some feedback on this!

**Comment by ahmaxed** [comment]
Looks good. I wonder if we should start left aligning the titles.
Have a look at the failing tests, so I could go for a deep dive review.
Isolating the stats into its own component makes sense.

**Comment by jenna5376** [comment]
Thank you so much for the feedback! I just addressed the failing tests.
I think we should left align the titles all at once in a separate pull request. What do you think?

File: client/src/components/profile/components/stats.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        </div>
+        <div>
+          <b>{t('profile.total-points')}</b>
+          <p className='stat'>{points}</p>
+        </div>
+        <div>
+          <b data-testid='longest-streak'>{t('profile.longest-streak')}</b>
+          <p className='stat'>{longestStreak || 0}</p>
+        </div>
+      </div>
```
Comment (bbsmooth) [inline client/src/components/profile/components/stats.tsx:130]: This feels like it should have a little more meaning semantically, such as putting these in a description list.

Comment (Sembauke) [reply client/src/components/profile/components/stats.tsx:130]:
```html 
<div className='stats-container'>
  <dl>
    <dt><b data-testid='current-streak'>{t('profile.current-streak')}</b></dt>
    <dd className='stat'>{currentStreak || 0}</dd>

    <dt><b>{t('profile.total-points')}</b></dt>
    <dd className='stat'>{points}</dd>

    <dt><b data-testid='longest-streak'>{t('profile.longest-streak')}</b></dt>
    <dd className='stat'>{longestStreak || 0}</dd>
  </dl>
</div>
```
Something like this.

**Review by Sembauke — APPROVED** [review]

File: client/src/components/profile/components/stats.test.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+describe('<Stats/>', () => {
+  it('calculates the correct longest streak', () => {
+    render(<Stats {...props} />);
+    expect(screen.getByTestId('longest-streak')).toHaveTextContent(
+      'profile.longest-streak'
+    );
+  });
+
+  it('calculates the correct current streak', () => {
```
Comment (shootermv) [inline client/src/components/profile/components/stats.test.tsx:18]: i have a feeling im missing something here: whta is the difference between 2 tests?

Comment (naomi-lgbt) [reply client/src/components/profile/components/stats.test.tsx:18]: Between this one and the "longest streak" test?

Comment (shootermv) [reply client/src/components/profile/components/stats.test.tsx:18]: ok, sorry, my bad - now i see: difference is `current streak` vs `longest streak`

**Review by ahmaxed — APPROVED** [review]

**Pull request merged**